include/deal.II/lac/la_vector.templates.h

```diff
@@ -398,7 +398,7 @@ namespace LinearAlgebra
 
     // in >> c;
     in.read(&c,1);
-    AssertThrow(c=']', ExcIO());
+    AssertThrow(c==']', ExcIO());
```

Comment:
@bangerth It is weird that when I find a typo or bug like this that, more often than not, the same issue exists elsewhere. Follow up PR en route.